### PR TITLE
Fix -Wtautological-constant-out-of-range-compare warnings on 32-bit systems

### PR DESCRIPTION
**Pull request by hebasto**
On 32-bit system
```
$ uname -nrmo
odroid-hc1 4.14.187-odroidxu4 armv7l GNU/Linux
$ lsb_release -ds
Ubuntu 20.04.1 LTS
$ clang --version
clang version 10.0.0-4ubuntu1 
Target: armv7l-unknown-linux-gnueabihf
Thread model: posix
InstalledDir: /usr/bin
```

the clang fires the following warnings:
```
  CXX      policy/libbitcoin_common_a-feerate.o
policy/feerate.cpp:14:20: warning: result of comparison of constant 9223372036854775807 with expression of type 'size_t' (aka 'unsigned int') is always true [-Wtautological-constant-out-of-range-compare]
    assert(nBytes_ <= uint64_t(std::numeric_limits<int64_t>::max()));
           ~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:27: note: expanded from macro 'assert'
     (static_cast <bool> (expr)                                         \
                          ^~~~
policy/feerate.cpp:25:20: warning: result of comparison of constant 9223372036854775807 with expression of type 'size_t' (aka 'unsigned int') is always true [-Wtautological-constant-out-of-range-compare]
    assert(nBytes_ <= uint64_t(std::numeric_limits<int64_t>::max()));
           ~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:27: note: expanded from macro 'assert'
     (static_cast <bool> (expr)                                         \
                          ^~~~
2 warnings generated.
```

Fix is safe as https://github.com/bitcoin/bitcoin/blob/a57af897ec16976b28de05aa0b9c3f6a96d73ede/src/compat/assumptions.h#L55-L59

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

Please consider enabling `-Wtautological-constant-out-of-range-compare` too :)

**Comment by hebasto** [comment]
> Please consider enabling `-Wtautological-constant-out-of-range-compare` too :)

Isn't it enabled by default?
https://clang.llvm.org/docs/DiagnosticsReference.html#wtautological-constant-out-of-range-compare

**Comment by theStack** [comment]
Code Review ACK 7c30e9f6f2f18057366383cb2b0a54d99731041c

File: src/policy/feerate.cpp [hunk 1]
```diff
@@ -20,7 +19,6 @@ CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nBytes_)
 
 CAmount CFeeRate::GetFee(size_t nBytes_) const
 {
-    assert(nBytes_ <= uint64_t(std::numeric_limits<int64_t>::max()));
```
Comment (MarcoFalke) [inline src/policy/feerate.cpp:23]: How is this `assert` save to remove for 64-bit?

**Comment by hebasto** [comment]
Updated 7c30e9f6f2f18057366383cb2b0a54d99731041c -> 3759a5b3be7a88f265a282388af603b27e458ec0 ([pr19735.01](https://github.com/hebasto/bitcoin/commits/pr19735.01) -> [pr19735.02](https://github.com/hebasto/bitcoin/commits/pr19735.02)):
- rebased
- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/19735#discussion_r625671778)

File: src/policy/feerate.cpp (hunk 1 above)
Comment (hebasto) [reply src/policy/feerate.cpp:23]: Oh, my bad!

Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/19735#issuecomment-834304120).

**Comment by hebasto** [comment]
The problem has been resolved in #21848 :tiger2:

**Pull request closed without merging**